app/models/user.rb

```diff
@@ -156,6 +156,31 @@ def mark_in_person_completion_survey_sent(issuer)
     nil
   end
 
+  def increment_second_factor_attempts_count!
+    User.transaction do
+      sql = <<-SQL
+      UPDATE users
+      SET second_factor_attempts_count = COALESCE(second_factor_attempts_count, 0) + 1,
+      updated_at = NOW(),
+      second_factor_locked_at = CASE
+        WHEN COALESCE(second_factor_attempts_count, 0) + 1 >= ?
+        THEN NOW()
+        ELSE NULL
+        END WHERE id = ? RETURNING second_factor_attempts_count, second_factor_locked_at;
+      SQL
+      query = User.sanitize_sql_array(
+        [sql,
+         IdentityConfig.store.login_otp_confirmation_max_attempts, self.id],
+      )
+      result = User.connection.execute(query)
+      self.second_factor_attempts_count = result.first.fetch('second_factor_attempts_count')
+      self.second_factor_locked_at = result.first.fetch('second_factor_locked_at')
```

Comment:
small DRY change idea
```suggestion
      result = User.connection.execute(query).first
      self.second_factor_attempts_count = result.fetch('second_factor_attempts_count')
      self.second_factor_locked_at = result.fetch('second_factor_locked_at')
```